### ISSUE TITLE
Ensure existing bookmarks work

### ISSUE THREAD

**Issue by JessicaMarine1** [GSA/smartpay-website#84]
As a user of the legacy system, I need any existing bookmark/link to still work, so that I can still easily access the new site.

Many users of the legacy system likely have bookmarks / links saved. We should ensure folks' bookmarks/links still work (e.g., by using redirects), or provide a workaround if we change the page URLs they're familiar with (e.g., have the 404 error page list the top requested content pages).

**Acceptance Criteria:**

- [ ]

**Comment by JessicaMarine1** [comment]
@A-L33 ^^^

**Comment by brentryanjohnson** [comment]
I'll work on this next (after accessibility testing).

Ideally, we would be able map chunks of redirects based on a hook in the existing paths, but most of the pages on the current site are in a `/content/` subdirectory, so they aren't currently organized by path.

I think we can start by using the content docs the team created, which contain the current URLs for the pages that were rewritten and consolidated. Those _should_ represent the most highly accessed pages, and I can map those to the new URL paths.

**Comment by jfredrickson5** [comment]
@brentryanjohnson Would it be possible to build a mapping of old pages to new from the content audit? With Astro, we can give it a list of old URLs and their respective new URLs, and it'll handle the rest.

**Comment by brentryanjohnson** [comment]
Yep! That's what I'm working on. I don't think we need to map everything over, but I'm crosswalking the pages we moved and consolidated to their origin.

**Comment by brentryanjohnson** [comment]
Here's a first attempt to map paths for redirects. @A-L33 Can you take a look and make sure nothing (important) is missing?

This approach doesn't contemplate redirecting _every_ page (that is, it doesn't identify an _individual_ page redirect for every existing page). If we need to go larger scale with redirects, we can, but we might want to take some time to think through which pages make sense. Outside of these specific page-level redirects, all other URLs would redirect to the new homepage.

New page | Redirect from
-- | --
about/about-gsa-smartpay | https://smartpay.gsa.gov/content/about-gsa-smartpay
about/benefits | https://smartpay.gsa.gov/gsa-smartpay-benefits
about/business-lines | https://smartpay.gsa.gov/purchase-program-overview-account-holdershttps://smartpay.gsa.gov/purchase-program-overview-aopcshttps://smartpay.gsa.gov/content/travel-program-overview-account-holdershttps://smartpay.gsa.gov/content/travel-program-overview-aopcshttps://smartpay.gsa.gov/content/fleet-program-overview-account-holdershttps://smartpay.gsa.gov/content/fleet-program-overviewhttps://smartpay.gsa.gov/content/integrated-program-overviewhttps://smartpay.gsa.gov/content/integrated
about/logos | https://smartpay.gsa.gov/content/logos-designs
about/master-contract | https://smartpay.gsa.gov/content/gsa-smartpay-master-contract
about/strategic-payment-solutions | https://smartpay.gsa.gov/content/strategic-payment-solutions-0 https://smartpay.gsa.gov/content/strategic-payment-solutions-1
about/refunds | https://smartpay.gsa.gov/content/refundshttps://smartpay.gsa.gov/gsa-smartpay-refunds
about/statistics-and-reports | https://smartpay.gsa.gov/content/program-statisticshttps://smartpay.gsa.gov/program-statistics-overviewhttps://smartpay.gsa.gov/content/sales-transactions-account-holder-datahttps://smartpay.gsa.gov/content/interesting-data-metricshttps://smartpay.gsa.gov/content/travel-card-spend-us-regionhttps://smartpay.gsa.gov/content/cfo-reportshttps://smartpay.gsa.gov/about-gsa-smartpay/program-statistics/savings-calculatorhttps://smartpay.gsa.gov/top-vendor-reportshttps://smartpay.gsa.gov/content/gsa-smartpay-purchase-card-statistics-reports-fpdshttps://smartpay.gsa.gov/gsa-smartpay-socioeconomic-statistics-reports
how-it-works/ | https://smartpay.gsa.gov/content/how-gsa-smartpay-works
how-it-works/eligibility | https://smartpay.gsa.gov/content/who-eligible
stakeholders/key-players | https://smartpay.gsa.gov/content/key-players-purchasing-process
stakeholders/program-coordinators | https://smartpay.gsa.gov/content/purchase-program-coordinator-responsibilities
stakeholders/account-holders | https://smartpay.gsa.gov/content/purchase-account-holder-responsibilities
merchants/merchant-overview | https://smartpay.gsa.gov/content/gsa-smartpay-vendorshttps://smartpay.gsa.gov/content/accepting-gsa-smartpay-payment-solutionhttps://smartpay.gsa.gov/content/card-driven-strategic-sourcing
smarttax/state-tax-forms | https://smartpay.gsa.gov/smarttaxhttps://smartpay.gsa.gov/content/specific-state-tax-informationhttps://smartpay.gsa.gov/content/about-smarttaxHttps://smartpay.gsa.gov/statetax/[state]
smarttax/state-tax-forms/[state] |  
smarttax/resources |  
smarttax/recognizing-your-account | https://smartpay.gsa.gov/content/recognizing-your-account
smarttax/legal-history | https://smartpay.gsa.gov/content/legal-history
policies-and-audits/policies | https://smartpay.gsa.gov/content/policieshttps://smartpay.gsa.gov/content/american-recovery-reinvestment-acthttps://smartpay.gsa.gov/content/federal-acquisition-regulationhttps://smartpay.gsa.gov/content/federal-travel-regulation-ftrhttps://smartpay.gsa.gov/content/omb-memorandum-m-17-26https://smartpay.gsa.gov/content/omb-circular-no-123-appendix-bhttps://smartpay.gsa.gov/content/omb-memorandum-m-13-21https://smartpay.gsa.gov/content/omb-memo-m-20-21-covid-reportinghttps://smartpay.gsa.gov/content/diem-rateshttps://smartpay.gsa.gov/ndaa-section-889https://smartpay.gsa.gov/content/public-law
policies-and-audits/audits | https://smartpay.gsa.gov/content/audit-repository
policies-and-audits/smart-bulletins | https://smartpay.gsa.gov/content/smart-bulletinshttps://smartpay.gsa.gov/smart-bulletin-no-037https://smartpay.gsa.gov/content/smart-bulletin-no-036https://smartpay.gsa.gov/content/smart-bulletin-no-035https://smartpay.gsa.gov/content/smart-bulletin-no-034https://smartpay.gsa.gov/content/smart-bulletin-no-033https://smartpay.gsa.gov/content/smart-bulletin-no-032https://smartpay.gsa.gov/content/smart-bulletin-no-031https://smartpay.gsa.gov/content/smart-bulletin-no-030https://smartpay.gsa.gov/content/smart-bulletin-no-029https://smartpay.gsa.gov/content/smart-bulletin-no-028https://smartpay.gsa.gov/content/smart-bulletin-no-027https://smartpay.gsa.gov/content/smart-bulletin-no-026https://smartpay.gsa.gov/content/smart-bulletin-no-025https://smartpay.gsa.gov/content/smart-bulletin-no-024https://smartpay.gsa.gov/content/smart-bulletin-no-023https://smartpay.gsa.gov/content/smart-bulletin-no-022https://smartpay.gsa.gov/content/smart-bulletin-no-021https://smartpay.gsa.gov/content/smart-bulletin-no-020https://smartpay.gsa.gov/content/smart-bulletin-no-019https://smartpay.gsa.gov/content/smart-bulletin-no-018https://smartpay.gsa.gov/smart-bulletin-no-017https://smartpay.gsa.gov/content/smart-bulletin-no-016https://smartpay.gsa.gov/content/smart-bulletin-no-015https://smartpay.gsa.gov/content/smart-bulletin-no-014https://smartpay.gsa.gov/content/smart-bulletin-no-013https://smartpay.gsa.gov/content/smart-bulletin-no-012https://smartpay.gsa.gov/content/smart-bulletin-no-011https://smartpay.gsa.gov/content/smart-bulletin-no-010https://smartpay.gsa.gov/content/smart-bulletin-no-009https://smartpay.gsa.gov/content/smart-bulletin-no-008https://smartpay.gsa.gov/content/smart-bulletin-no-007https://smartpay.gsa.gov/content/smart-bulletin-no-006https://smartpay.gsa.gov/content/smart-bulletin-no-005https://smartpay.gsa.gov/content/smart-bulletin-no-004https://smartpay.gsa.gov/content/smart-bulletin-no-003https://smartpay.gsa.gov/content/smart-bulletin-no-002https://smartpay.gsa.gov/content/smart-bulletin-no-001
  |  
  |  
resources/tools/section-889 | https://smartpay.gsa.gov/section-889-easy-search-tools
resources/publications/ | https://smartpay.gsa.gov/content/general-resources-account-holdershttps://smartpay.gsa.gov/content/additional-resources-aopcshttps://smartpay.gsa.gov/content/general-resources-aopcshttps://smartpay.gsa.gov/content/additional-resources-account-holdershttps://smartpay.gsa.gov/content/additional-resources-vendors
resources/publications/travel | https://smartpay.gsa.gov/content/travel-resources-account-holdershttps://smartpay.gsa.gov/content/travel-resources-aopcs
resources/publications/purchase | https://smartpay.gsa.gov/content/purchase-resources-account-holdershttps://smartpay.gsa.gov/purchase-resources-aopcs
resources/publications/fleet | https://smartpay.gsa.gov/content/fleet-resources-account-holdershttps://smartpay.gsa.gov/content/fleet-manager-responsibilitieshttps://smartpay.gsa.gov/content/fleet-resources-account-holdershttps://smartpay.gsa.gov/content/gsa-smartpay-fleet-account
resources/publications/all-business-lines |  
resources/publications/videos | https://smartpay.gsa.gov/content/multimedia
events/gsa-smartpay-forum | https://smartpay.gsa.gov/2023-gsa-smartpay-training-forumhttps://smartpay.gsa.gov/forumhttps://smartpay.gsa.gov/2023-gsa-smartpay-training-forum-presentations-virtual-forum
faq/ | https://smartpay.gsa.gov/content/common-questions-public-law-112-194-and-omb-mem-m-13-21https://smartpay.gsa.gov/helphttps://smartpay.gsa.gov/account-holder-common-questionshttps://smartpay.gsa.gov/content/vendor-common-questionshttps://smartpay.gsa.gov/common-questions-about-state-taxeshttps://smartpay.gsa.gov/aopc-common-questionshttps://smartpay.gsa.gov/frequently-asked-questions-faqs-federal-government-shutdown
contact/ | https://smartpay.gsa.gov/gsa-smartpay-program-support
contact/banks-contact | https://smartpay.gsa.gov/content/agency-bank-informationhttps://smartpay.gsa.gov/content/bank-customer-service-centers

**Comment by brentryanjohnson** [comment]
The formatting of those ☝️ got a little funky, so you might want to view them in sheet 2 of the [🔒 IA google sheet](https://docs.google.com/spreadsheets/d/1-nIXpGYsCgKcrXA4LLTEECBILCZOVWmn4n4abvFJGZU/edit?usp=sharing)

**Comment by A-L33** [comment]
@brentryanjohnson Can we add https://smartpay.gsa.gov/glossary -> Master Contract?  Thank you!

**Comment by brentryanjohnson** [comment]
Yep! It's kind of difficult to see, but the table includes `https://smartpay.gsa.gove/statetax/[state]`-->`smarttax/state-tax-forms`

I'll add the glossary redirecting to Master Contract!

**Comment by A-L33** [comment]
Thank you Ryan!  I'm good here @rebekahperillo if you are.

**Comment by rebekahperillo** [comment]
@A-L33 Trusting your work here. Good to go. :) Thank you!

**Comment by jfredrickson5** [comment]
@brentryanjohnson I was able to implement most of these so far, but added comments to the [IA sheet](https://docs.google.com/spreadsheets/d/1-nIXpGYsCgKcrXA4LLTEECBILCZOVWmn4n4abvFJGZU/edit?usp=sharing) where I need clarification on a few rows. Could you take a look whenever you have a chance? No rush, because I also have the accessibility changes to implement in the meantime.

**Comment by brentryanjohnson** [comment]
@jfredrickson5 I replied in the sheet!

**Comment by JessicaMarine1** [comment]
@brentryanjohnson Jeff indicated that this should be done. Can you please review before we send to POs?

**Comment by brentryanjohnson** [comment]
I corrected some destination paths in #285. It should be ready for review once that branch is merged in.

**Comment by JessicaMarine1** [comment]
@A-L33 @rebekahperillo this is ready for review. @brentryanjohnson any suggestions how they might review this?

**Comment by brentryanjohnson** [comment]
@A-L33 @rebekahperillo We won't _really_ be able to test this until we're in production, since the redirects won't be active until there is no longer a viable link for a current page (that is, until the current https://smartpay.gsa.gov/content/about-gsa-smartpay becomes the new smartpay.gsa.gov/about – which isn't a live URL yet!).

In the meantime, the logic that will ultimately drive redirects is [in this file](https://github.com/GSA/smartpay-website/blob/staging/src/config/redirects.js).

Beyond the first few lines, you'll see "key:value" pairs. Basically, one URL path, then a comma, then another URL path.

```
  "/content/about-gsa-smartpay": "/about",
  "/gsa-smartpay-benefits": "/about/benefits",
```

The first path will be the page on the _current_ site. Then a comma, followed by the _new_ page we're building.

These should look like what you [reviewed in a previous comment](https://github.com/GSA/smartpay-website/issues/84#issuecomment-1661007063) ☝️, but this is the code version that will drive the actual redirects. 👍 

Let me know if you have any questions!

**Comment by JessicaMarine1** [comment]
Maybe we move it (or add a new issue) to post-MVP and test once we're live. We have a similar issue with DAP.

**Comment by jfredrickson5** [comment]
You can sort of test this currently by tacking on the "from" redirect to the preview URL, like this:

Take the base preview URL, which is:

https://federalist-ab31a10d-375d-4040-9324-1ae94e8a36b9.sites.pages.cloud.gov/site/gsa/smartpay-website/

And add, for example, `content/about-gsa-smartpay` (which is [this page on the existing site](https://smartpay.gsa.gov/content/about-gsa-smartpay)) to the end of it, to get:

https://federalist-ab31a10d-375d-4040-9324-1ae94e8a36b9.sites.pages.cloud.gov/site/gsa/smartpay-website/content/about-gsa-smartpay

And if you follow that link, it should redirect you to the new About page at:

https://federalist-ab31a10d-375d-4040-9324-1ae94e8a36b9.sites.pages.cloud.gov/site/gsa/smartpay-website/about/

Hope this helps!

**Comment by brentryanjohnson** [comment]
@rebekahperillo I added these to a 🔒 [new google sheet](https://docs.google.com/spreadsheets/d/1ZNVAjsy96CtDvHnMkjre0ESKpcOjy_hRmAP49x7-dVQ/edit?usp=sharing) and activated the (legacy) link paths to try to make it easier for you to test them 👍.

**Comment by rebekahperillo** [comment]
You are awesome. Thank you!

Rebekah Knouse Perillo
Business Management Specialist
Center for Charge Card Management
Office of Professional Services and Human Capital Categories
Federal Acquisition Service, U.S. General Services Administration
301-938-5257
